examples/c_api/quickstart_dimension_labels.c

```diff
@@ -0,0 +1,534 @@
+/**
+ * @file quickstart_dimension_labels.c
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * When run, this program will create a simple 2D dense array with 3 dimension
+ * labels, write data to it, and read back data from the array and the labels.
+ * If the array already exists, then the writing and creation steps will be
+ * skipped.
+ *
+ * Array Summary:
+ *  * Array Type: Dense
+ *  * Dimensions:
+ *    - x: (type=INT32, domain=[0, 5])
+ *    - sample: (type=INT32, domain=[0,3])
+ *
+ *  * Attributes:
+ *    - a: (type=INT16)
+ *
+ *  * Labels on dimension 'x':
+ *    - x (order=INCREASING, type=FLOAT64)
+ *    - y (order=INCREASING, type=FLOAT64)
+ *
+ *  * Labels on dimension 'sample':
+ *    - timestamp (order=INCREASING, type=DATETIME_SEC)
+ *
+ * Data:
+ *  a = [ 1,  2,  3,  4,  5,  6,
+ *        7,  8,  9, 10, 11, 12,
+ *       13, 14, 15, 16, 17, 18,
+ *       19, 20, 21, 22, 23, 23]
+ *  x = [-1.0, -0.6, -0.2, 0.2, 1.0]
+ *  y = [0.0, 2.0, 4.0, 6.0, 8.0, 10.0]
+ *  timestamp = [8:52:23, 8:59:40, 9:12:11, 9:13:48]
+ */
+
+#include <stdio.h>
+#include <tiledb/tiledb.h>
+#include <tiledb/tiledb_experimental.h>
+
+/** Macro that will end a function on a TileDB error.*/
+#define RETURN_IF_ERROR(rc) \
```

Comment:
Not for this PR, just noting that we need to create a little header-only helper library for the examples (cc @teo-tsirpanis who did something similar recently).